presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestWriter.java

```diff
@@ -232,6 +232,63 @@ public void testCreateBucketSortedTableAsSelect()
         }
     }
 
+    @Test
+    public void testRecoverableGroupedExecution()
+    {
+        Session session = buildSessionForTableWrite();
+
+        String sourceTable = generateRandomTableName();
+        String sinkTable = generateRandomTableName();
+
+        getQueryRunner().execute(session, String.format(
+                "CREATE TABLE %s WITH (format = '%s', " +
+                        "partitioned_by = ARRAY[ 'orderstatus' ], " +
+                        "bucketed_by=array['custkey'], " +
+                        "bucket_count=17) " +
+                        "AS SELECT custkey, orderkey, orderstatus FROM orders", sourceTable, "DWRF"));
+        assertQuery(
+                String.format("SELECT custkey, orderkey, orderstatus FROM %s", sourceTable),
+                "SELECT custkey, orderkey, orderstatus FROM orders");
+
+        try {
+            for (String tableFormat : TABLE_FORMATS) {
+                try {
+                    String groupBy = String.format(
+                            "SELECT custkey, min(orderkey) orderkey, min(orderstatus) orderstatus " +
+                                    "FROM %s " +
+                                    "GROUP BY custkey", sourceTable);
+
+                    String ctas = String.format(
+                            "CREATE TABLE %s WITH (format = '%s', " +
+                                    "partitioned_by = ARRAY[ 'orderstatus' ], " +
+                                    "bucketed_by=array['custkey'], " +
+                                    "bucket_count=17) " +
+                                    "AS " +
+                                    "%s", sinkTable, tableFormat, groupBy);
+                    String ctasPlan = getQueryRunner().execute(session, "EXPLAIN (TYPE DISTRIBUTED) " + ctas).getOnlyValue().toString();
```

Comment:
nit: Would it be helpful to extract a helper method that takes a query and return "EXPLAIN (TYPE DISTRIBUTED) <query>" results?